File: evidence/generated-systems/angular/fintech-on-angular/src/app/models/TradeOrder.js

// Define collection and schema for TradeOrder
export interface TradeOrder {
    orderId:
	type : string
    quantity:
	type : String
    limitPrice:
	type : Money
    placedAt:
	type : DateTime
    Portfolio:
	type : Schema.Types.ObjectId
    Security:
	type : Schema.Types.ObjectId
    Trades:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Trade' }]
    Side:
 	type : String
    Type:
 	type : String
    Status:
 	type : String
    TimeInForce:
 	type : String
#
    collection: 'tradeOrders'
}

File: evidence/generated-systems/rubyonrails/fintech-on-rails/fintechonrails/app/models/tradeorder.rb
class TradeOrder < ApplicationRecord
  enum Side: [:Buy, :Sell]
  enum Type: [:Market, :Limit, :Stop, :StopLimit]
  enum Status: [:New, :PartiallyFilled, :Filled, :Cancelled, :Rejected, :Expired]
  enum TimeInForce: [:Day, :GTC, :IOC, :FOK]


  composed_of :money,
    class_name: "Money",
    mapping: [
      ${$mapping}, 
      %w[money_currency currency]
    ]

  composed_of :dateTime,
    class_name: "DateTime",
    mapping: [
      %w[dateTime_value value]
    ]

  has_many :Portfolio, class_name: 'InvestmentPortfolio'
  has_many :Security, class_name: 'Security'
  has_many :Trades, class_name: 'Trade'

end
